api/src/main/java/org/apache/cloudstack/acl/RoleType.java

```diff
@@ -68,26 +80,15 @@ public static RoleType fromMask(int mask) {
         return Unknown;
     }
 
-    public static RoleType getByAccountType(final short accountType) {
-        RoleType roleType = RoleType.Unknown;
-        switch (accountType) {
-            case Account.ACCOUNT_TYPE_ADMIN:
-                roleType = RoleType.Admin;
-                break;
-            case Account.ACCOUNT_TYPE_DOMAIN_ADMIN:
-                roleType = RoleType.DomainAdmin;
-                break;
-            case Account.ACCOUNT_TYPE_RESOURCE_DOMAIN_ADMIN:
-                roleType = RoleType.ResourceAdmin;
-                break;
-            case Account.ACCOUNT_TYPE_NORMAL:
-                roleType = RoleType.User;
-                break;
+    public static RoleType getByAccountType(final Account.Type accountType) {
+        RoleType t = ACCOUNT_TYPE_MAP.get(accountType);
+        if (t == null) {
```

Comment:
if you use `Type.UNKNOWN` rather than null then this can be simplified to

`return (t == UNKNOWN) ? RoleType.UNKNOWN : t`

Reply:
I prefer not to use contracted ifs, I believe it worsens the legibility for no gain whatsoever.

Reply:
its not about `if` here but its about not using `null`

Reply:
The problem is, if a `Account.Type` is passed that is not in the map, `ACCOUNT_TYPE_MAP.get(accountType)` will return null nonetheless, for example, passing Account.Type.PROJECT will return null. So checking `if (t == null)` is still needed.